### DISCUSSION TITLE
Publish to WinGet (Automated)

### DISCUSSION THREAD

**Discussion by vedantmgoyal9** [ppy/osu#18274] [Ideas] [answered]
* Follow-up of https://github.com/ppy/osu/issues/9073

It was two years ago when WinGet was just released. Now, 2 years later, things have changed.

Seeing the main issue was continuously updating manifests with every new release, there is now a GitHub Action which automatically publishes releases to the WinGet repository using a GitHub personal access token.

If you want I can create a pull request that will add the action, and would need no maintenance (everything's automated ;))

**Comment by bdach** [comment]
Seeing that literally nobody has inquired about this for the 2 years since the original thread, I'd say we're fine to continue ignoring winget's existence and keep our one single distribution method for windows as is.

**Comment by vedantmgoyal9** [comment]
* https://github.com/ppy/osu/discussions/15007

@peppy @bdach There are people who are interested but seems like it didn't gather much attention from the maintainers, so people also didn't care about discussing this topic. I've created this discussion so that it may get into attention of other maintainers and hopefully we can make osu! available to different platforms via package managers. I would suggest WinGet for Windows and brew for macOS and Linux(?).

If you want I can create GitHub Workflows for it so that you don't have to spend time manually publishing updates to the above.

**Reply by peppy** [reply]
Please wait for other users to actually show support. I've never heard of a single person using winget in the wild.

**Reply by bdach** [reply]
I might have been wrong about "literally nobody" asking about it but 2 users in 2 years is still pretty insignificant, all things considered. Discussions are there to vote on them to show support, and that one barely got any support.

**Comment by vedantmgoyal9** [comment, marked as answer]
I would more happy if you would pin it.

**Comment by SheepCommander** [comment]
cri ;-;

Well osu! is on winget now, and it pulls the latest release from this repo sorta...

**Reply by bdach** [reply]
Dear lord........ https://github.com/microsoft/winget-pkgs/pull/133933

@peppy for your awareness i guess. people literally cannot stop themselves doing this over and over without even a courtesy attempt of contacting us.

@Exorcism0666 can you elaborate what this is about? why was there no attempt to contact us? are you aware that the game contains an auto-update facility? that presumably will break whatever versioning scheme this winget thing is attempting to enforce?

**Reply by Exorcism0666** [reply]
@bdach I don't have to contact you if I feel like it, now if this really is a problem for you, please open an “Issue” in the Winget repo to let us know if you want to remove osu.

**Reply by peppy** [reply]
jesus.

**Reply by BiosPlus** [reply]
I mean, is there really an expectation to contact you about redistribution of the software even though the practice is in line with your [license](https://github.com/ppy/osu/blob/master/LICENCE) 🤔

**Reply by peppy** [reply]
licence != trademark. also common decency. also it was published with completely incorrect naming / metadata.

**Reply by bdach** [reply]
also [this](https://github.com/microsoft/winget-pkgs/issues/152768) is a thing that also happened?

@Susko3 why did _you_ not even backlink that issue to this thread or mention this to any of us over here? what in the heck is going on here? i expect much better from you specifically as a recurring contributor.
